lib/hammer_cli_foreman/sessions.rb

```diff
@@ -88,5 +91,17 @@ def destroy
     def valid?
       !id.nil? && !id.empty?
     end
+
+    def auth_type_oauth?
+      @auth_type == AUTH_TYPES[:oauth_authentication_code_grant] ||
+        @auth_type == AUTH_TYPES[:oauth_password_grant]
+    end
+
+    def expired?
+      return false unless auth_type_oauth?
+      return false unless (Time.parse(token_expires_at) - Time.now.utc).to_i.negative?
+
+      true
+    end
```

Comment:
How about:
```ruby
def expired?
  return false unless auth_type_oauth?
  return true if token_expires_at.to_s.empty? # we don't support empty expiration dates or assume the expiration date is  now
  return false unless (Time.parse(token_expires_at.to_s).utc - Time.now.utc).to_i.negative? # convert all time to UTC and then compare

  true
end
```

Reply:
I think the second condition here, `we don't support empty expiration dates or assume the expiration date is  now` gets handled by the third condition. I mean if the token_expires_at is not present then in the third condition the `(Time.parse(token_expires_at.to_s).utc - Time.now.utc).to_i.negative?` will be true. Finally, the function will return true. 

Let me know if still think the condition must be added?